go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -0,0 +1,1768 @@
+	return eIPC.retryEgressIPs.WatchResource()
+}
+
+func (eIPC *egressIPController) Stop() {
+	close(eIPC.stopChan)
+	eIPC.wg.Wait()
+	if eIPC.egressNodeHandler != nil {
+		eIPC.watchFactory.RemoveNodeHandler(eIPC.egressNodeHandler)
+	}
+	if eIPC.egressIPHandler != nil {
+		eIPC.watchFactory.RemoveEgressIPHandler(eIPC.egressIPHandler)
+	}
+	if eIPC.cloudPrivateIPConfigHandler != nil {
+		eIPC.watchFactory.RemoveCloudPrivateIPConfigHandler(eIPC.cloudPrivateIPConfigHandler)
+	}
+}
+
+// egressIPClusterControllerEventHandler object handles the events
+// from retry framework.
+type egressIPClusterControllerEventHandler struct {
+	objretry.EventHandler
+	objType  reflect.Type
+	eIPC     *egressIPController
+	syncFunc func([]interface{}) error
+}
+
+// egressIPClusterControllerEventHandler functions
+
+// AddResource adds the specified object to the cluster according to its type and
+// returns the error, if any, yielded during object creation.
+func (h *egressIPClusterControllerEventHandler) AddResource(obj interface{}, fromRetryLoop bool) error {
+	var err error
+	switch h.objType {
+	case factory.EgressNodeType:
+		node := obj.(*kapi.Node)
+		if err = h.eIPC.setupNodeForEgress(node); err != nil {
+			return err
+		}
+		nodeEgressLabel := util.GetNodeEgressLabel()
+		nodeLabels := node.GetLabels()
+		_, hasEgressLabel := nodeLabels[nodeEgressLabel]
+		if hasEgressLabel {
+			h.eIPC.setNodeEgressAssignable(node.Name, true)
+		}
+		isReady := h.eIPC.isEgressNodeReady(node)
+		if isReady {
+			h.eIPC.setNodeEgressReady(node.Name, true)
+		}
+		isReachable := h.eIPC.isEgressNodeReachable(node)
+		if hasEgressLabel && isReachable && isReady {
+			h.eIPC.setNodeEgressReachable(node.Name, true)
+			if err := h.eIPC.addEgressNode(node.Name); err != nil {
+				return err
+			}
+		}
+	case factory.EgressIPType:
+		eIP := obj.(*egressipv1.EgressIP)
+		return h.eIPC.reconcileEgressIP(nil, eIP)
+	case factory.CloudPrivateIPConfigType:
+		cloudPrivateIPConfig := obj.(*ocpcloudnetworkapi.CloudPrivateIPConfig)
+		return h.eIPC.reconcileCloudPrivateIPConfig(nil, cloudPrivateIPConfig)
+	default:
+		return fmt.Errorf("no add function for object type %s", h.objType)
+	}
+	return nil
+}
+
+// UpdateResource updates the specified object in the cluster to its version in newObj according
+// to its type and returns the error, if any, yielded during the object update.
+// The inRetryCache boolean argument is to indicate if the given resource is in the retryCache or not.
+func (h *egressIPClusterControllerEventHandler) UpdateResource(oldObj, newObj interface{}, inRetryCache bool) error {
+	switch h.objType {
+	case factory.EgressIPType:
+		oldEIP := oldObj.(*egressipv1.EgressIP)
+		newEIP := newObj.(*egressipv1.EgressIP)
+		return h.eIPC.reconcileEgressIP(oldEIP, newEIP)
+	case factory.EgressNodeType:
+		oldNode := oldObj.(*kapi.Node)
+		newNode := newObj.(*kapi.Node)
+		// Initialize the allocator on every update,
```

Comment:
yup, the change in nodeIP's is relevant only for the default LRPs we create for EIP at priority 102. Since that needs to be done from ovnkube-controller (previously master) we don't need this here.